IoT live tests fail during the "Remove test resources" since it could not find the S_RESOURCE_GROUP env var

The step is failing on all 3 OSes (Linux, Windows, MacOS).
cc @danewalton 

It looks like it might have been introduced due to this change, by @benbp:
https://github.com/Azure/azure-sdk-for-c/pull/2175

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1691344&view=logs&j=a9c6f0ff-98f8-542d-8584-a52715ad8a35&t=b239ef7e-7184-56f2-144f-34f3cc51b920

```text
Starting: Remove test resources
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.200.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_work\_temp\ef48bfdc-cd88-443f-a7b2-b0f5ddd6e294.ps1'"
Using module Az with version 7.3.2.
VERBOSE: Populating RepositorySourceLocation property for module Az.
11:59:19 AM - Logging into service principal '***'
VERBOSE: Running registered exit actions.
VERBOSE: Logging out of service principal '***'
D:\a\_work\1\s\eng\common\TestResources\Remove-TestResources.ps1 : Could not find resource group name environment variable 'S_RESOURCE_GROUP'. This is likely due to an earlier failure in the 'Deploy Test Resources' step above.
At D:\a\_work\_temp\ef48bfdc-cd88-443f-a7b2-b0f5ddd6e294.ps1:9 char:1
+ eng/common/TestResources/Remove-TestResources.ps1 `
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
+ FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Remove-TestResources.ps1
##[error]PowerShell exited with code '1'.
Finishing: Remove test resources
```


This pipeline has been failing with a warning for some time now. @danieljurek how come it manifests itself as a warning rather than an error, in this view, and within the pipeline view on AzDo?
![image](https://user-images.githubusercontent.com/6527137/177812328-7014ba52-d09e-484a-88af-76bf064adc30.png)

https://dev.azure.com/azure-sdk/internal/_build?definitionId=724&_a=summary

But, it does correctly highlight the error here. I don't think these pipeline run errors triggered email notifications.
![image](https://user-images.githubusercontent.com/6527137/177815386-41271a9f-8a95-4713-aa8f-afe2df827d46.png)

--- a/eng/pipelines/templates/jobs/archetype-sdk-tests.yml
+++ b/eng/pipelines/templates/jobs/archetype-sdk-tests.yml
@@ -98,4 +98,4 @@ jobs:
 
       - template: /eng/common/TestResources/remove-test-resources.yml
         parameters:
-          ServiceDirectory: '.'
+          ServiceDirectory: 'samples'

